src/game.cpp

```diff
@@ -11485,6 +11485,33 @@ bool game::walk_move( const tripoint &dest_loc )
     }
     u.set_underwater(false);
 
+    bool fire_removed = false;
+    const tripoint furn_pos = u.pos() + u.grab_point;
+    const tripoint furn_dest = dest_loc + u.grab_point;
+    const int fire_age = m.get_field_age( furn_pos, fd_fire );
+    int fire_str = 0;
+    const int fire_str_at_dest = m.get_field_strength( furn_dest, fd_fire );
+
+    const bool canmove = (
+    m.passable( furn_dest ) &&
+    critter_at<npc>( furn_dest ) == nullptr &&
+    critter_at<monster>( furn_dest ) == nullptr &&
+    ( !m.has_floor( furn_dest ) || m.has_flag( "FLAT", furn_dest ) ) &&
+    ( !m.has_floor( furn_dest ) || m.tr_at( furn_dest ).is_null() ) &&
+    !m.has_furn( furn_dest ) &&
+    m.veh_at( furn_dest ) == nullptr
+    );
```

Comment:
This whole thing is huge and most likely redundant. Big, redundant checks are very fragile.

I checked the surrounding code and it looks like you're executing this all in the wrong place: `grabbed_furn_move` should be doing all of those checks.
The only thing you may want to replace here should be the `prompt_dangerous_tile` part. It won't be pretty, but I think the simplest way to handle it would be to add "ignore small fires" argument to `prompt_dangerous_tile` and then in the function change
```
if( u.is_dangerous_field( e.second ) && fields_here.findField( e.first ) == nullptr ) {
```
to
```
if( u.is_dangerous_field( e.second ) && fields_here.findField( e.first ) == nullptr &&
( !ignore_small_fires || e.first != fd_fire || e.second.getFieldDensity() != 1 ) ) {
```

This would be messy, but not as messy as replicating a whole big check here and assuming this check will not change in the future revisions.